packages/calling/src/CallingClient/line/line.test.ts

```diff
@@ -0,0 +1,185 @@
+import * as regUtils from '../registration/register';
+
+describe('Line Tests', () => {
+  const mutex = new Mutex();
+  const webex = getTestUtilsWebex();
+  SDKConnector.setWebex(webex);
+
+  const defaultServiceData = {indicator: ServiceIndicator.CALLING, domain: ''};
+  const createRegistrationSpy = jest.spyOn(regUtils, 'createRegistration');
+
+  const mobiusUris = filterMobiusUris(getMobiusDiscoveryResponse(), URL);
+  const primaryMobiusUris = jest.fn(() => mobiusUris.primary);
+  const backupMobiusUris = jest.fn(() => mobiusUris.backup);
+  const userId = webex.internal.device.userId;
+  const clientDeviceUri = webex.internal.device.url;
+
+  const handleErrorSpy = jest.spyOn(utils, 'handleRegistrationErrors');
+
+  jest.clearAllMocks();
+
+  describe('Registration tests', () => {
+    let line;
+    const mockRegistrationBody = getMockDeviceInfo();
+
+    const discoveryBody: MobiusServers = getMobiusDiscoveryResponse();
+    const primaryUrl = `${discoveryBody.primary.uris[0]}/calling/web/`;
+
+    const registrationPayload = <WebexRequestPayload>(<unknown>{
+      statusCode: 200,
+      body: mockRegistrationBody,
+    });
+
+    const mockBody = {
+      userId: webex.internal.device.userId,
+      clientDeviceUri: webex.internal.device.url,
+      serviceData: {
+        domain: '',
+        indicator: 'calling',
+      },
+    };
+
+    beforeEach(() => {
+      line = new Line(
+        userId,
+        clientDeviceUri,
+        LineStatus.ACTIVE,
+        mutex,
+        primaryMobiusUris(),
+        backupMobiusUris(),
+        LOGGER.INFO
+      );
+    });
+
+    afterEach(() => {
+      jest.clearAllTimers();
+      jest.clearAllMocks();
+      jest.useRealTimers();
+    });
+
+    it('verify successful Registration cases and keepalive', async () => {
+      jest.useFakeTimers();
+      webex.request.mockReturnValue(registrationPayload);
+
+      expect(createRegistrationSpy).toBeCalledWith(
+        webex,
+        defaultServiceData,
+        expect.any(Mutex),
+        expect.anything(),
+        LOGGER.INFO
+      );
+      expect(line.getRegistrationStatus()).toEqual(MobiusStatus.DEFAULT);
+      await line.register();
+
+      expect(webex.request).toBeCalledTimes(1);
+      expect(handleErrorSpy).not.toBeCalled();
+
+      expect(line.getRegistrationStatus()).toEqual(MobiusStatus.ACTIVE);
+      expect(line.getActiveMobiusUrl()).toEqual(primaryUrl);
+      expect(line.getLoggingLevel()).toEqual(LOGGER.INFO);
+      expect(line.getDeviceId()).toEqual(mockRegistrationBody.device.deviceId);
+
+      /** trigger sendKeepAlive and verify whether keepalive request is sent or not */
+      webex.request.mockClear();
+
+      /* specify keepalive interval as 30 seconds and advance timers by 30 */
+      line.sendKeepAlive({...mockRegistrationBody, keepaliveInterval: 30});
+
+      jest.advanceTimersByTime(30 * 1000);
+      await Promise.resolve();
+
+      expect(webex.request).toBeCalledWith({
```

Comment:
Same comment here as well. Instead of toBeCalledWith use toBeCalledOnceWith, applies everywhere

Reply:
Done at all places.